python/cuml/ensemble/randomforest.pyx

```diff
@@ -181,7 +180,14 @@ cdef class RandomForest_impl():
         cdef cumlHandle* handle_ =\
             <cumlHandle*><size_t>self.handle.getHandle()
 
-        unique_labels = cupy.unique(y)
+        try:
+            import cupy as cp
+            unique_labels = cp.unique(y)
+        except ImportError:
+            warnings.warn("Using NumPy for number of class detection,"
+                          "install CuPy for faster processing.")
+            unique_labels = np.unique(y.copy_to_host())
```

Comment:
Do we end up with uniques on the host in this case and on the device in the other case? Maybe nicer to make them consistent?

Reply:
very true, and I'm ok with changing it, but the `unique_labels` variable is actually not used except to check that the labels are consecutive in `if i not in unique_labels:`. I wonder if its worth it to transfer them back to the device for that? What do you think?

Reply:
Looks like they should both be on host and perhaps a set to be honest since we're iterating over them and checking for membership, but I don't want to overcomplicate this... I'm ok to push as is and improve in the next pass